### ISSUE TITLE
quantity exception when passing value like 0.5

### ISSUE THREAD

**Issue by magamine** [darryldecode/laravelshoppingcart#200]
i need to pass float value like 0.1 , 0.2 and min 1 in validation is causing an exception

**Comment by lukaszokaj** [comment]
I solved it only in such a way that in the file 
"Project Name\vendor\darryldecode\cart\src\Darryldecode\Cart\Cart.php
in function:
`protected function validate($item)
    {
        $rules = array(
            'id' => 'required',
            'price' => 'required|numeric',
            'quantity' => 'required|numeric|min:1',
            'name' => 'required',
        );

        $validator = CartItemValidator::make($item, $rules);

        if ($validator->fails()) {
            throw new InvalidItemException($validator->messages()->first());
        }

        return $item;
    }`

I deleted the condition "min:1" for quantity
so I have:
`protected function validate($item)
    {
        $rules = array(
            'id' => 'required',
            'price' => 'required|numeric',
            'quantity' => 'required|numeric',
            'name' => 'required',
        );

        $validator = CartItemValidator::make($item, $rules);

        if ($validator->fails()) {
            throw new InvalidItemException($validator->messages()->first());
        }

        return $item;
    }`

Now it is working

**Comment by irvingzamora** [comment]
No, because in methods updateQuantityNotRelative and updateQuantityRelative yuo have a (int) casting
